File: backend/src/modules/transportation/excel/excel.controller.ts
import { Body, Controller, Get, Post, UseGuards } from '@nestjs/common';
import { HeaderAuthGuard } from '../../../common/auth/header-auth.guard';
import { ExcelService } from './excel.service';

// Base path /api/Transportation (faithful to the documented URLs, §6.13); every
// action requires the CompanyName + UserToken headers via the guard.
@Controller('api/Transportation')
@UseGuards(HeaderAuthGuard)
export class ExcelController {
  constructor(private readonly service: ExcelService) {}

  @Get('downloadExcelUserWithRoutesTemplete')
  downloadUserWithRoutesTemplate() {
    return this.service.downloadUserWithRoutesTemplate();
  }

  @Get('downloadExcelUserActiveTemplete')
  downloadUserActiveTemplate() {
    return this.service.downloadUserActiveTemplate();
  }

  /** GET downloadExcelUsersList — export current passengers as .xlsx (base64). */
  @Get('downloadExcelUsersList')
  usersListExcel() {
    return this.service.usersListExcel();
  }

  @Post('InsertUsersWithRoutesExcel')
  insertUsersWithRoutesExcel(@Body() body: { File?: string }) {
    return this.service.insertUsersWithRoutesExcel(body?.File);
  }

  @Post('InsertUserNotActiveExcel')
  insertUserNotActiveExcel(@Body() body: { File?: string }) {
    return this.service.insertUserNotActiveExcel(body?.File);
  }
}

File: backend-dotnet/GarasTransport.Api/Modules/Transportation/Excel/ExcelController.cs
using GarasTransport.Api.Common;
using Microsoft.AspNetCore.Mvc;

namespace GarasTransport.Api.Modules.Transportation.Excel;

// Base path /api/Transportation (§6.13). Every action requires CompanyName +
// UserToken headers via the filter.
[Route("api/Transportation")]
[HeaderAuth]
public class ExcelController : ApiControllerBase
{
    private readonly ExcelService _service;

    public ExcelController(ExcelService service) => _service = service;

    [HttpGet("downloadExcelUserWithRoutesTemplete")]
    public Task<object> DownloadUserWithRoutesTemplate() => _service.DownloadUserWithRoutesTemplate();

    [HttpGet("downloadExcelUserActiveTemplete")]
    public Task<object> DownloadUserActiveTemplate() => _service.DownloadUserActiveTemplate();

    [HttpGet("downloadExcelUsersList")]
    public Task<object> UsersListExcel() => _service.UsersListExcel();

    [HttpPost("InsertUsersWithRoutesExcel")]
    public Task<object> InsertUsersWithRoutesExcel([FromBody] FileBody body) =>
        _service.InsertUsersWithRoutesExcel(body?.File);

    [HttpPost("InsertUserNotActiveExcel")]
    public Task<object> InsertUserNotActiveExcel([FromBody] FileBody body) =>
        _service.InsertUserNotActiveExcel(body?.File);
}

public class FileBody
{
    public string? File { get; set; }
}
